Marlin/src/gcode/feature/trinamic/M911-M915.cpp

```diff
@@ -161,8 +161,8 @@ void GcodeSuite::M912() {
   void GcodeSuite::M913() {
     #define TMC_SAY_PWMTHRS(A,Q) tmc_get_pwmthrs(stepper##Q, TMC_##Q, planner.axis_steps_per_mm[_AXIS(A)])
     #define TMC_SET_PWMTHRS(A,Q) tmc_set_pwmthrs(stepper##Q, value, planner.axis_steps_per_mm[_AXIS(A)])
-    #define TMC_SAY_PWMTHRS_E(E) do{ const uint8_t extruder = E; tmc_get_pwmthrs(stepperE##E, TMC_E##E, planner.axis_steps_per_mm[E_AXIS_N]); }while(0)
-    #define TMC_SET_PWMTHRS_E(E) do{ const uint8_t extruder = E; tmc_set_pwmthrs(stepperE##E, value, planner.axis_steps_per_mm[E_AXIS_N]); }while(0)
+    #define TMC_SAY_PWMTHRS_E(E) do{tmc_get_pwmthrs(stepperE##E, TMC_E##E, planner.axis_steps_per_mm[E_AXIS_N]); }while(0)
+    #define TMC_SET_PWMTHRS_E(E) do{tmc_set_pwmthrs(stepperE##E, value, planner.axis_steps_per_mm[E_AXIS_N]); }while(0)
```

Comment:
`E_AXIS_N` needs `extruder` to be defined when `DISTINCT_E_FACTORS` is enabled, so this can't be left out. However, an `UNUSED(extruder)` could be added to the end of each of these blocks to suppress the warning, without affecting the case where `DISTINCT_E_FACTORS` is enabled.

Reply:
> E_AXIS_N needs extruder to be defined when DISTINCT_E_FACTORS is enabled, so this can't be left out.

Oh, I see. I was blindsided by the macros.

Reply:
Also, just curious, why the do{}while(0)? Isn't this just equivalent to plain old curly braces? Like this:

```
#define TMC_SAY_PWMTHRS_E(E) {const uint8_t extruder = E; tmc_get_pwmthrs(stepperE##E, TMC_E##E, planner.axis_steps_per_mm[E_AXIS_N]); }
```

Reply:
Ah. I was not aware of older compilers complaining about the semicolon at the end of blocks. The idiomatic use of do/while makes a lot more sense now. Thank you for the explanation.